sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -98,29 +116,71 @@ private static Mono<okhttp3.Request> toOkHttpRequest(HttpRequest request) {
             return Mono.just(requestBuilder.head().build());
         }
 
-        return toOkHttpRequestBody(request.getBody(), request.getHeaders())
+        return toOkHttpRequestBody(request.getBodyAsBinaryData(), request.getHeaders())
             .map(okhttpRequestBody -> requestBuilder.method(request.getHttpMethod().toString(), okhttpRequestBody)
                 .build());
     }
 
     /**
-     * Create a Mono of okhttp3.RequestBody from the given java.nio.ByteBuffer Flux.
+     * Create a Mono of okhttp3.RequestBody from the given BinaryData.
      *
-     * @param bbFlux stream of java.nio.ByteBuffer representing request content
+     * @param bodyContent The request body content
      * @param headers the headers associated with the original request
-     * @return the Mono emitting okhttp3.RequestBody
+     * @return the Mono emitting okhttp request
      */
-    private static Mono<RequestBody> toOkHttpRequestBody(Flux<ByteBuffer> bbFlux, HttpHeaders headers) {
-        Mono<okio.ByteString> bsMono = bbFlux == null
-            ? EMPTY_BYTE_STRING_MONO
-            : toByteString(bbFlux);
+    private Mono<RequestBody> toOkHttpRequestBody(BinaryData bodyContent, HttpHeaders headers) {
+        String contentType = headers.getValue("Content-Type");
+        MediaType mediaType = (contentType == null) ? null : MediaType.parse(contentType);
+
+        if (bodyContent == null) {
+            return Mono.just(RequestBody.create(ByteString.EMPTY, mediaType));
```

Comment:
For well-known content types, is it worth creating empty `RequestBody` constants and reusing them instead of creating them each time?

Reply:
yup, and it seems we can do it for any media type (it's optional request body property as it turns out).